### PR TITLE
WIP: First setup temporal data

### PR DESCRIPTION
**Pull request by rduivenvoorde**
I received some example data from Nyall and am planning to add it to the cookbook.

Thinking about asking @nyalldawson for his earthquacke examples he showed in his webinar (for those who havent't seen it: https://www.youtube.com/watch?v=vgDg5cRwPRw) 
If others have better idea's for temporal data to use in Documentation (Manual AND cookbook) please let us know.

### CODE REVIEW

**Comment by havatv** [comment]
> If others have better idea's for temporal data to use in Documentation (Manual AND cookbook) please let us know.

Not sure if it is better, but trajectory data (x, y, t) or (x, y, z, t) for moving objects are common (but perhaps too complex?).
Trajectories could be for animals, vehicles, goods or people (if there are no privacy issues). Regarding privacy issues - it could be possible to obfuscate the data.
Temporal data for spatially static objects are probably less complex. Could be climate data or other kinds of environmental data from measurement stations or remote sensing time series data.

**Comment by nyalldawson** [comment]
There's also an "all in one" export-to-multiple-images function available now. I would like to tweak the API of this over the next week and then will post an example of it here!

File: docs/pyqgis_developer_cookbook/temporal_data.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+       img = QImage(map_settings.outputSize(), map_settings.outputImageFormat())
+       img.fill(map_settings.backgroundColor().rgb())
+
+       p = QPainter()
+       p.begin(img)
+
+       map_settings.setIsTemporal(True)
+       map_settings.setTemporalRange(time_range)
+
+      render = QgsMapRendererCustomPainterJob(map_settings, p)
```
Comment (havatv) [inline docs/pyqgis_developer_cookbook/temporal_data.rst:46]:
```suggestion
       render = QgsMapRendererCustomPainterJob(map_settings, p)
```

File: docs/pyqgis_developer_cookbook/temporal_data.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      render.start()
```
Comment (havatv) [inline docs/pyqgis_developer_cookbook/temporal_data.rst:47]:
```suggestion
       render.start()
```

File: docs/pyqgis_developer_cookbook/temporal_data.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      render.waitForFinished()
```
Comment (havatv) [inline docs/pyqgis_developer_cookbook/temporal_data.rst:48]:
```suggestion
       render.waitForFinished()
```

File: docs/pyqgis_developer_cookbook/temporal_data.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      p.end()
```
Comment (havatv) [inline docs/pyqgis_developer_cookbook/temporal_data.rst:49]:
```suggestion
       p.end()
```

File: docs/pyqgis_developer_cookbook/temporal_data.rst [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
```
Comment (havatv) [inline docs/pyqgis_developer_cookbook/temporal_data.rst:50]:
```suggestion
```

File: docs/pyqgis_developer_cookbook/temporal_data.rst [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      return img
```
Comment (havatv) [inline docs/pyqgis_developer_cookbook/temporal_data.rst:51]:
```suggestion
       return img
```

File: docs/pyqgis_developer_cookbook/temporal_data.rst [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+.. contents::
+   :local:
+
+Since QGIS 3.14 QGIS has native tools to handle Temporal Data. 
+There is a Temporal data navigation Widget to configure QGIS, and in PyQGIS there
+is the QgsTemporalNavigationObject
+
+.. testcode:: temporal_data
+
```
Comment (havatv) [inline docs/pyqgis_developer_cookbook/temporal_data.rst:35]:
```suggestion

    from qgis.core import QgsTemporalNavigationObject
```
Or something like that - to make travis happy.

**Comment by rduivenvoorde** [comment]
@havatv sorry for not working with your comment, but that first draft was really preliminary.

This commit has working code. Not sure if I should write more info into this.

@nyalldawson do you think the 'write a serie of images to tmp dir' is too much? I could not find the "all in one" export-to-multiple-images function you were talking about :-)
To update the actual gui (when starting to 'play' the navigator) we will some even probably? So it shows up from 'idle' to the right visual state I mean?

File: docs/pyqgis_developer_cookbook/temporal_data.rst [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+    iface = start_qgis()
+
+
+.. highlight:: python
+   :linenothreshold: 5
+
+The code snippets on this page need the following imports if you're outside the pyqgis console:
+
+.. testcode:: expr
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/temporal_data.rst:12]: one of the main reasons of the failure
```suggestion
.. testcode:: temporal_data
```

**Review by DelazJ — COMMENTED** [review]
This should fix some travis issues.
the world_map opening warning does not break build. it pops up every time (eg https://travis-ci.org/github/qgis/QGIS-Documentation/builds/738801257#L925). Having it fixed would not be bad though.

**Comment by nyalldawson** [comment]
@rduivenvoorde 
> do you think the 'write a serie of images to tmp dir' is too much? I could not find the "all in one" export-to-multiple-images function you were talking about :-)

Can't you use `QgsTemporalUtils.exportAnimation`?

**Comment by rduivenvoorde** [comment]
Sorry for the git mess...

**Comment by rduivenvoorde** [comment]
@elpaso (if you have time to have a look, thanks) do you maybe have an idea why these tests keep failing? Travis is complaining that `iface.mapCanvas().temporalController()` is returning None, but it just works in the Python Console (with an empty project).
Is there something I am missing here to get it to work?

**Comment by DelazJ** [comment]
@rduivenvoorde I force-pushed your branch to trigger current CI. Let's see...

**Comment by DelazJ** [comment]
@rduivenvoorde Close this?

**Pull request closed without merging**